src/platform/tests/TestKeyValueStoreMgr.cpp

```diff
@@ -0,0 +1,233 @@
+{
+    CHIP_ERROR err;
+    const char * kTestKey   = "array_key";
+    uint32_t kTestValue[5] = { 1, 2, 3, 4, 5 };
+    uint32_t read_value[5];
+    size_t read_size;
+    err = KeyValueStoreMgr().Put(kTestKey, kTestValue);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    err = KeyValueStoreMgr().Get(kTestKey, read_value, sizeof(read_value), &read_size);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    // Verify if read value is the same as wrote one
+    NL_TEST_ASSERT(inSuite, memcmp(kTestValue, read_value, sizeof(kTestValue)) == 0);
+    NL_TEST_ASSERT(inSuite, read_size == sizeof(kTestValue));
+    err = KeyValueStoreMgr().Delete(kTestKey);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    // Try to get deleted key and verify if CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND is returned
+    err = KeyValueStoreMgr().Get(kTestKey, read_value, sizeof(read_value), &read_size);
+    NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+}
+
+static void TestKeyValueStoreMgr_StructKey(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err;
+    struct TestStruct
+    {
+        uint8_t value1;
+        uint32_t value2;
+    };
+    const char * kTestKey   = "struct_key";
+    TestStruct kTestValue{ value1 : 1, value2 : 2 };
+    TestStruct read_value;
+    size_t read_size;
+    err = KeyValueStoreMgr().Put(kTestKey, kTestValue);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    err = KeyValueStoreMgr().Get(kTestKey, &read_value, sizeof(read_value), &read_size);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    // Verify if read value is the same as wrote one
+    NL_TEST_ASSERT(inSuite, kTestValue.value1 == read_value.value1);
+    NL_TEST_ASSERT(inSuite, kTestValue.value2 == read_value.value2);
+    NL_TEST_ASSERT(inSuite, read_size == sizeof(kTestValue));
+    err = KeyValueStoreMgr().Delete(kTestKey);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    // Try to get deleted key and verify if CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND is returned
+    err = KeyValueStoreMgr().Get(kTestKey, &read_value, sizeof(read_value), &read_size);
+    NL_TEST_ASSERT(inSuite, err == CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND);
+}
+
+static void TestKeyValueStoreMgr_UpdateKeyValue(nlTestSuite * inSuite, void * inContext)
+{
+    CHIP_ERROR err;
+    const char * kTestKey = "update_key";
+    uint32_t read_value;
+    size_t read_size;
+    for (uint32_t i = 0; i < 10; i++)
+    {
+        err = KeyValueStoreMgr().Put(kTestKey, i);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+        err = KeyValueStoreMgr().Get(kTestKey, &read_value);
+        NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+        NL_TEST_ASSERT(inSuite, i == read_value);
+    }
+    err = KeyValueStoreMgr().Delete(kTestKey);
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+}
+
+static void TestKeyValueStoreMgr_MultiReadKey(nlTestSuite * inSuite, void * inContext)
```

Comment:
Since it may not be easy to implement read-from-offset on our platform, consider adding another test case which simply tests that read with a buffer smaller than the stored data returns CHIP_ERROR_BUFFER_TOO_SMALL, but the buffer contains part of the data.

Reply:
Done, as the TestKeyValueStoreMgr_TooSmallBufferRead.